tests/lava/proc/dense/test_models.py

```diff
@@ -553,3 +553,484 @@ def test_bitacc_pm_recurrence(self):
         rcfg = DenseRunConfig(select_tag='fixed_pt')
         dense.run(condition=rcnd, run_cfg=rcfg)
         dense.stop()
+
+
+class TestDelayDenseProcessModel(unittest.TestCase):
+    """Tests for ProcessModels of Dense with synaptic delay."""
+
+    def test_float_pm_buffer_delay(self):
+        """Tests floating point Dense ProcessModel connectivity and temporal
+        dynamics. All input 'neurons' from the VecSendandRcv fire
+        once at time t=4, and only 1 connection weight
+        in the Dense Process is non-zero. The value of the delay matrix for
+        this weight is 2. The non-zero connection should have an activation of
+        1 at timestep t=7.
+        """
+        shape = (3, 4)
+        num_steps = 8
+        # Set up external input to emulate every neuron spiking once on
+        # timestep 4
+        vec_to_send = np.ones((shape[1],), dtype=float)
+        send_at_times = np.repeat(False, (num_steps,))
+        send_at_times[3] = True
+        sps = VecSendandRecvProcess(shape=(shape[1],), num_steps=num_steps,
+                                    vec_to_send=vec_to_send,
+                                    send_at_times=send_at_times)
+        # Set up Dense Process with a single non-zero connection weight at
+        # entry [2, 2] of the connectivity matrix and a delay of 2 at entry
+        # [2, 2] in the delay matrix.
+        weights = np.zeros(shape, dtype=float)
+        weights[2, 2] = 1
+        delays = np.zeros(shape, dtype=int)
+        delays[2, 2] = 2
+        dense = DelayDense(weights=weights, delays=delays)
+        # Receive neuron spikes
+        spr = VecRecvProcess(shape=(num_steps, shape[0]))
+        sps.s_out.connect(dense.s_in)
+        dense.a_out.connect(spr.s_in)
+        # Configure execution and run
+        rcnd = RunSteps(num_steps=num_steps)
+        rcfg = DenseRunConfig(select_tag='floating_pt')
+        dense.run(condition=rcnd, run_cfg=rcfg)
+        # Gather spike data and stop
+        spk_data_through_run = spr.spk_data.get()
+        dense.stop()
+        # Gold standard for the test
+        # a_out will be equal to 1 at timestep 7, because the dendritic
+        #  accumulators work on inputs from the previous timestep + 2.
+        expected_spk_data = np.zeros((num_steps, shape[0]))
+        expected_spk_data[6, 2] = 1.
```

Comment:
The equation you stated is not correct, since the sum is 7 :P
We send at time step 4, at time step 5 the spikes arrive without delay. 
Not sure what the dend buffer delay would be, I just added the delay to the expected values from the dense unit tests.

Reply:
Ah that's correct.